gcloud/pubsub/_helpers.py

```diff
@@ -31,15 +39,27 @@ def topic_name_from_path(path, project):
              the project from the ``path`` does not agree with the
              ``project`` passed in.
     """
-    # PATH = 'projects/%s/topics/%s' % (PROJECT, TOPIC_NAME)
-    path_parts = path.split('/')
-    if (len(path_parts) != 4 or path_parts[0] != 'projects' or
-            path_parts[2] != 'topics'):
-        raise ValueError('Expected path to be of the form '
-                         'projects/{project}/topics/{topic_name}')
-    if (len(path_parts) != 4 or path_parts[0] != 'projects' or
-            path_parts[2] != 'topics' or path_parts[1] != project):
-        raise ValueError('Project from client should agree with '
-                         'project from resource.')
-
-    return path_parts[3]
+    return _name_from_project_path(path, project, _TOPIC_PATH_TEMPLATE)
+
+
+_SUBSCRIPTION_PATH_TEMPLATE = re.compile(
+    r'projects/(?P<project>\w+)/subscriptions/(?P<name>\w+)')
```

Comment:
These are private details, not public exports:  I actually prefer them nearer the point of use.  In fact, I think this is a case for dropping the two-line spacing requirement:  I'm only making them globals to speed up execution of the functions they are helpers for.  Ideally they would be inlined into the function (maybe we could do that anyway, and skip compiling them, relying on the `re` module's cache for speed).